### PR TITLE
:bug: Update the practitioner details to expect string resource Ids

### PR DESCRIPTION
**Pull request by dubdabasoduba**
fixes #https://github.com/opensrp/fhircore/issues/1995

### CODE REVIEW

**Comment by dubdabasoduba** [comment]
Please review this too once you are done https://github.com/opensrp/hapi-fhir-jpaserver-starter/pull/56

File: practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java [hunk 1]
```diff
@@ -109,12 +112,11 @@ public PractitionerDetails getPractitionerDetails(
...
+            String practitionerIdStringValue;
             boolean isIdNumeric = false;
             if (practitioner.getIdElement() != null
                     && practitioner.getIdElement().getIdPart() != null
-                    && StringUtils.isNumeric(
-                            ((IBaseResource) practitioner).getIdElement().getIdPart())) {
+                    && StringUtils.isNumeric(practitioner.getIdElement().getIdPart())) {
                 practitionerId =
                         practitioner.getIdElement() != null
                                         && practitioner.getIdElement().getIdPart() != null
```
Comment (ndegwamartin) [inline practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java]: We already have the null check in the outer condition as well as the default of `0L`, we can refactor this line to just be   `practitionerId = practitioner.getIdElement().getIdPartAsLong();`

File: practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java [hunk 2]
```diff
@@ -126,15 +128,12 @@ public PractitionerDetails getPractitionerDetails(
                 practitionerIdStringValue =
                         practitioner.getIdElement() != null
                                         && practitioner.getIdElement().getIdPart() != null
-                                ? ((IBaseResource) practitioner)
-                                        .getIdElement()
-                                        .getIdPart()
-                                        .toString()
+                                ? practitioner.getIdElement().getIdPart()
                                 : "";
```
Comment (ndegwamartin) [inline practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java]: We already have the null check in the outer condition as well as the default of `""`, we can refactor this line to just be   `practitionerIdStringValue = practitioner.getIdElement().getIdPart();`

File: practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java [hunk 3]
```diff
@@ -109,12 +112,11 @@ public PractitionerDetails getPractitionerDetails(
             IBaseResource practitioner =
                     practitioners.size() > 0 ? practitioners.get(0) : new Practitioner();
             Long practitionerId = null;
-            String practitionerIdStringValue = null;
+            String practitionerIdStringValue;
```
Comment (ndegwamartin) [inline practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java]: We can set the defaults to `0L` and `""` respectively to coincide with the recommended refactors below

File: practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java (hunk 2 above)
Comment (dubdabasoduba) [reply practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java]: Makes sense, actually, I don't think we need to care about whether the id is a string or numeric.  https://github.com/opensrp/hapi-fhir/blob/148cbf119b3a90d129850ff19fc39ae3fdddc6c0/hapi-fhir-base/src/main/java/org/hl7/fhir/instance/model/api/IIdType.java#L50 returns a string which will work on search so we do not need to check whether the id is numeric or not

**Review by ndegwamartin — APPROVED** [review]

**Pull request merged**